File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {

        int maxGain = 0;
        int sum = 0;

        int front = 0;
        int back = 0;

        for (int i = 0; i < prices.length; i++) {
            sum = prices[front] - prices[back];
            if (sum > maxGain){
                maxGain = sum;
            }

            if (prices[front] < prices[back]){
                back = front;
            }

            front++;

        }

        return maxGain;
            
        
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        front = 0
        back = len(heights) - 1

        maxArea = min(heights[front],heights[back]) * back - front
        while (front != back):
            if (heights[front] <= heights[back]):
                front += 1
                if (min(heights[front],heights[back]) * (back - front) > maxArea):
                    maxArea = min(heights[front],heights[back]) * (back - front)
            if (heights[front] > heights[back]):
                back -= 1
                if (min(heights[front],heights[back]) * (back - front) > maxArea):
                    maxArea = min(heights[front],heights[back]) * (back - front)
            
        return maxArea